zerver/views/streams.py

```diff
@@ -380,6 +383,20 @@ def update_stream_backend(
     if stream_post_policy is not None:
         do_change_stream_post_policy(stream, stream_post_policy, acting_user=user_profile)
 
+    if can_remove_subscribers_group_id is not None:
+        if sub is None and stream.invite_only:
+            # Admins cannot change this setting for unsubscribed private streams.
+            raise JsonableError(_("Invalid stream ID"))
+
+        user_group = access_user_group_for_setting(
+            can_remove_subscribers_group_id,
+            user_profile,
+            setting_name="can_remove_subscribers_group",
+            require_system_group=True,
+        )
```

Comment:
It's a bit annoying that this is duplicated with the similar copy in stream creation; I'm not sure what a good pattern for avoiding that would be, but I'm a bit worried about in the future parameters like `require_system_group` being managed incorrectly for them.

I think our idea of having those live in `@dataclass` structures or something is probably a pretty good one.

If we define those data structures nicely, we can also potentially include them in the API, and have the web app read them; so that the function of the form `get_realm_user_groups_for_dropdown_list_widget` could use the exact same configuration (without duplication) to know which groups to display... so that the frontend wouldn't require any changes at all if we were to adjust the configuration to say, start allowing non-system groups for this setting.